data-quality-service/src/main/scala/com/hashmap/dataquality/metadata/MetadataDao.scala

```diff
@@ -0,0 +1,39 @@
+package com.hashmap.dataquality.metadata
+
+import org.rocksdb.util.SizeUnit
+import org.rocksdb.{CompactionStyle, CompressionType, Options, RocksDB}
+import org.springframework.stereotype.Service
+
+@Service
+class MetadataDao {
+
+  private val UTF8: String = "UTF-8"
+  private val dbFilePath: String = "/tmp/db"
+
+  RocksDB.loadLibrary()
+
+  private val options: Options = new Options().setCreateIfMissing(true)
+    .setCreateIfMissing(true)
+    .setWriteBufferSize(200 * SizeUnit.MB)
+    .setMaxWriteBufferNumber(3)
+    .setMaxBackgroundCompactions(10)
+    .setCompressionType(CompressionType.SNAPPY_COMPRESSION)
+    .setCompactionStyle(CompactionStyle.UNIVERSAL)
+
+  private val db: RocksDB = RocksDB.open(dbFilePath)
+
+  //TODO: Handle faults/exceptions
```

Comment:
DB connection is not getting close. It should get close when application shuts down. Check @PreDestory annotation. Exception handling is also remaining.